### PR TITLE
Try using a percentage based calculation for block drag scrolling velocity.

### PR DESCRIPTION
**Pull request by talldan**
## Description
This tries an iteration on the behavior in #23082. 

See also #23446 for an alternative.

It's based on the refactor to hooks in #23444, so the diff looks bigger than it is. That PR should ideally be merged first.

#### How does this differ from #23082

In #23082, when the user drags more than 50px the page starts scrolling in the direction they've dragged. The speed of the scrolling is based on how far the user drags.

Because the scroll speed is calculated using pixels, if the user starts dragging a block that's already near the top of the screen upwards there's a limit to the speed that the page can scroll. On the other hand, when dragging downwards from the top, very fast scroll speeds can be unexpectedly achieved due to the amount of space a user can drag into.

This PR makes the calculation percentage based. If the user drags to the top of the page the page scrolls at the same speed as if they dragged to the bottom, regardless of where they started dragging from.

## How has this been tested?
1. Create a post with lots of blocks
2. Try dragging blocks to an off-screen position.

## Types of changes
Non-breaking enhancement

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jasmussen** [comment]
Before:

before

After:

after

Thank you for making this PR! 

The scroll while dragging enhancement is great, but it was definitely a little aggressive. This balances it out, way better. 👍 👍

**Comment by ellatrix** [comment]
Nice!

**Comment by gravityrail** [comment]
Very nice improvement. Thanks!

File: packages/block-editor/src/components/block-draggable/use-scroll-when-dragging.js [hunk 1]
```diff
@@ -60,15 +60,42 @@ export default function useScrollWhenDragging( dragElement ) {
...
+				offsetDragStartPosition - SCROLL_INACTIVE_DISTANCE_PX,
+				0
+			);
+			const dragDistance = Math.max(
+				offsetDragStartPosition -
+					offsetDragPosition -
+					SCROLL_INACTIVE_DISTANCE_PX,
+				0
+			);
+			const distancePercentage = dragDistance / moveableDistance;
```
Comment (gravityrail) [inline packages/block-editor/src/components/block-draggable/use-scroll-when-dragging.js:97]: These lines appear to be duplicates of lines in the previous if-clause - is there a way to eliminate this duplication?

**Review by gravityrail — CHANGES_REQUESTED** [review]
A little feedback on duplication, otherwise looks good

**Pull request merged**